### PR TITLE
ENH: support writing the geoarrow-based encodings of GeoParquet

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
~WIP, on top of https://github.com/geopandas/geopandas/pull/3219~

~Will probably break out the reading side separately as a first PR, since that does not rely on https://github.com/geopandas/geopandas/pull/3219~ (reading side: https://github.com/geopandas/geopandas/pull/3278)

This implements the writing side for geoarrow encodings in GeoParquet (https://github.com/geopandas/geopandas/issues/3253)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,7 +50,8 @@ New features and improvements:
   the specified columns (#3101).
 - Added support to ``read_file`` for the ``mask`` keyword for the pyogrio engine (#3062).
 - Added support to ``read_file`` for the ``columns`` keyword for the fiona engine (#3133).
-- Added support to ``read_parquet`` for reading files using the GeoArrow-based native geometry encoding of GeoParquet 1.1 (#3253).
+- Added support to ``to_parquet`` and ``read_parquet`` for writing and reading files
+  using the GeoArrow-based native geometry encoding of GeoParquet 1.1 (#3253).
```
Comment (martinfleis) [inline CHANGELOG.md:54]:
```suggestion
  using the GeoArrow-based native geometry encoding of GeoParquet 1.1 (#3253, #3275).
```

**Review by martinfleis — COMMENTED** [review]
Looks fine, with my limited GeoArrow knowledge. Some parts of the code are reported as uncovered but at least the ValueError has an explicit test, so I am not sure how trustworthy it is here.

**Comment by jorisvandenbossche** [comment]
> Some parts of the code are reported as uncovered but at least the ValueError has an explicit test, so I am not sure how trustworthy it is here.

Yeah, codecov is doing a bit strange the last days .. Most of the annotations I see are definitely covered

**Comment by kylebarron** [comment]
How are you thinking of handling the fact that GeoParquet 1.1 hasn't been released yet? Are you thinking of holding this until 1.1 is released?

**Comment by jorisvandenbossche** [comment]
I was planning to add an "experimental" note to the docstring (it's all opt-in), but yes ideally we would finalize the GeoParquet spec soon (same issue exists for GDAL ..)

**Comment by jorisvandenbossche** [comment]
(the same issue is true for the bbox writing/reading, so was planning to do that as a follow-up)

We could also add a warning to make it more visible, but given it is purely opt-in a note in the docs might be sufficient?

**Comment by jorisvandenbossche** [comment]
Going to merge this one giving there are a few other PRs that touch the same code

**Pull request merged**